diff --git a/examples/pxScene2d/external/breakpad-chrome_55/src/client/linux/microdump_writer/microdump_writer.cc b/examples/pxScene2d/external/breakpad-chrome_55/src/client/linux/microdump_writer/microdump_writer.cc
index 6f5b435..a508667 100644
--- a/examples/pxScene2d/external/breakpad-chrome_55/src/client/linux/microdump_writer/microdump_writer.cc
+++ b/examples/pxScene2d/external/breakpad-chrome_55/src/client/linux/microdump_writer/microdump_writer.cc
@@ -571,7 +571,7 @@ class MicrodumpWriter {
 
   void* Alloc(unsigned bytes) { return dumper_->allocator()->Alloc(bytes); }
 
-  const struct ucontext* const ucontext_;
+  const ucontext_t* const ucontext_;
 #if !defined(__ARM_EABI__) && !defined(__mips__)
   const google_breakpad::fpstate_t* const float_state_;
 #endif
