diff --git a/examples/pxScene2d/external/zlib-1.2.11/zconf.h b/examples/pxScene2d/external/zlib-1.2.11/zconf.h
index 5e1d68a..77398c1 100644
--- a/examples/pxScene2d/external/zlib-1.2.11/zconf.h
+++ b/examples/pxScene2d/external/zlib-1.2.11/zconf.h
@@ -431,11 +431,11 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if 1    /* was set to #if 1 by ./configure */
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if 1    /* was set to #if 1 by ./configure */
 #  define Z_HAVE_STDARG_H
 #endif
 
