diff --git a/src/client/linux/dump_writer_common/ucontext_reader.h b/src/client/linux/dump_writer_common/ucontext_reader.h
index b6e77b4b5..61855688b 100644
--- a/src/client/linux/dump_writer_common/ucontext_reader.h
+++ b/src/client/linux/dump_writer_common/ucontext_reader.h
@@ -37,6 +37,10 @@
 #include "common/memory.h"
 #include "google_breakpad/common/minidump_format.h"
 
+#ifndef ucontext
+#define ucontext ucontext_t
+#endif
+
 namespace google_breakpad {
 
 // Wraps platform-dependent implementations of accessors to ucontext structs.
diff --git a/src/client/linux/handler/exception_handler.h b/src/client/linux/handler/exception_handler.h
index 591c31085..acf65bd40 100644
--- a/src/client/linux/handler/exception_handler.h
+++ b/src/client/linux/handler/exception_handler.h
@@ -44,6 +44,10 @@
 #include "common/using_std_string.h"
 #include "google_breakpad/common/minidump_format.h"
 
+#ifndef ucontext
+#define ucontext ucontext_t
+#endif
+
 namespace google_breakpad {
 
 // ExceptionHandler
