Index: nanosvg/src/nanosvgrast.h
===================================================================
--- nanosvg.orig/src/nanosvgrast.h
+++ nanosvg/src/nanosvgrast.h
@@ -1132,11 +1132,11 @@ static void nsvg__rasterizeSortedEdgesFu
 	int y, s;
 	int e = 0;
 	int maxWeight = (255 / NSVG__SUBSAMPLES);  // weight per vertical scanline
-	int xmin, xmax;
+	int xmin = 0, xmax = 0;
 
 	for (y = 0; y < r->height; y++) {
 		memset(r->scanline, 0, r->width);
-		xmin = r->width;
+		xmin = (r->width > 0) ? r->width : 0;
 		xmax = 0;
 		for (s = 0; s < NSVG__SUBSAMPLES; ++s) {
 			// find center of pixel for this scanline
