From 10742394f676dd3ea324e6b9f4ad165f029361bf Mon Sep 17 00:00:00 2001
From: madanagopalt <madanagopal123@gmail.com>
Date: Mon, 18 Dec 2017 11:26:17 +0530
Subject: [PATCH] fix for animation in oscillate mode

---
 examples/pxScene2d/src/pxScene2d.cpp | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/examples/pxScene2d/src/pxScene2d.cpp b/examples/pxScene2d/src/pxScene2d.cpp
index f6ea0ce..1a1b34f 100644
--- a/examples/pxScene2d/src/pxScene2d.cpp
+++ b/examples/pxScene2d/src/pxScene2d.cpp
@@ -641,7 +641,7 @@ rtError pxObject::animateToObj(rtObjectRef props, double duration,
     for (uint32_t i = 0; i < len; i++)
     {
       rtString key = keys.get<rtString>(i);
-      animateToInternal(key, props.get<float>(key), duration, ((pxConstantsAnimation*)CONSTANTS.animationConstants.getPtr())->getInterpFunc(interp), (pxConstantsAnimation::animationOptions)options, count,(i==0)?promise:rtObjectRef(),animateObj);
+      animateToInternal(key, props.get<float>(key), duration, ((pxConstantsAnimation*)CONSTANTS.animationConstants.getPtr())->getInterpFunc(interp), (pxConstantsAnimation::animationOptions)options, count, (i==0)?promise:rtObjectRef(),animateObj);
     }
   }
   if (NULL != animateObj.getPtr())
@@ -1002,25 +1002,27 @@ void pxObject::update(double t)
     t1 = t1-t2; // 0-1
 
     double d = a.interpFunc(t1);
-    
     float from = a.from;
     float   to = a.to;
 
     if (a.options & pxConstantsAnimation::OPTION_OSCILLATE)
     {
+      bool justReverseChange = false;
+      double toVal = a.to;
       if( (fmod(t2,2) != 0))  // TODO perf chk ?
       {
         if(!a.reversing)
         {
           a.reversing = true;
+          justReverseChange = true;
           a.actualCount++;
         }
         from = a.to;
         to   = a.from;
-
       }
       else if( a.reversing && (fmod(t2,2) == 0))
       {
+        justReverseChange = true;
         a.reversing = false;
         a.actualCount++;
         a.start = -1;
@@ -1028,11 +1030,17 @@ void pxObject::update(double t)
       // Prevent one more loop through oscillate
       if(a.count != pxConstantsAnimation::COUNT_FOREVER && a.actualCount >= a.count )
       {
+          if (true == justReverseChange)
+          {
+            mCancelInSet = false;
+            set(a.prop, toVal);
+            mCancelInSet = true;
+          }
+
         if (NULL != animObj)
         {
           animObj->setStatus(pxConstantsAnimation::STATUS_ENDED);
         }
-
         cancelAnimation(a.prop, false, false, true);
 
         if (NULL != animObj)
@@ -1051,7 +1059,6 @@ void pxObject::update(double t)
     mCancelInSet = false;
     set(a.prop, v);
     mCancelInSet = true;
-
     if (NULL != animObj)
     {
       animObj->update(a.prop, &a, pxConstantsAnimation::STATUS_INPROGRESS);
-- 
1.9.1

